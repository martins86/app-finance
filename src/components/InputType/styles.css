.input-primary {
  padding: 16px 20px;
  width: 100%;
  background-color: #f0f2f5;
  border: none;
  border-radius: 5px;
  font-size: 16px;
}
