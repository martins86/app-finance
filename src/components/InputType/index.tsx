import './styles.css';

export interface InputParams {
  type: string;
  id?: string;
  nameField?: string;
  dataTestId: string;
  className?: string;
  placeholder?: string;
  value?: string;
  min?: string;
  onChange?: any;
  onKeyDown?: any;
  autoComplete?: string;
  ariaRequired?: boolean;
  ariaLabel?: string;
  defaultChecked?: boolean;
  required?: boolean;
  disabled?: boolean;
}

const InputType = ({
  type,
  id,
  nameField,
  dataTestId,
  className,
  placeholder,
  value,
  min,
  onChange,
  onKeyDown,
  autoComplete,
  ariaRequired,
  ariaLabel,
  required,
  disabled,
  defaultChecked,
}: InputParams) => {
  return (
    <input
      type={type}
      id={id}
      name={nameField}
      data-testid={dataTestId}
      className={className}
      placeholder={placeholder}
      value={value}
      min={min}
      onChange={onChange}
      onKeyDown={onKeyDown}
      autoComplete={autoComplete}
      aria-label={ariaLabel}
      aria-required={ariaRequired}
      required={required}
      disabled={disabled}
      defaultChecked={defaultChecked}
    />
  );
};

export default InputType;
