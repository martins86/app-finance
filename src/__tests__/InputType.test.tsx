import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router } from 'react-router-dom';

import InputType from '../components/InputType';

describe('Test <InputType />', () => {
  test('Should have a value of InputTextValue', () => {
    // Arrange
    const fn = jest.fn();
    render(
      <Router>
        <InputType
          type="text"
          dataTestId="teste-id"
          onChange={fn}
          placeholder="Any Text"
          value="Any Value"
        />
      </Router>
    );

    // Act
    const elInputText: HTMLInputElement =
      screen.getByPlaceholderText(/Any Text/i);

    // Assert
    expect(elInputText.value).toBe('Any Value');
  });

  test('Should call onChange function on each key pressed', () => {
    // Arrange
    const fn = jest.fn();
    render(
      <Router>
        <InputType
          type="text"
          dataTestId="teste-id"
          onChange={fn}
          placeholder="Any Text"
        />
      </Router>
    );

    // Act
    const elInputText: HTMLInputElement =
      screen.getByPlaceholderText(/Any Text/i);
    const valueOnInput = 'Any Value';

    userEvent.type(elInputText, valueOnInput);

    // Assert
    expect(elInputText.value).toBe(valueOnInput);
    expect(fn).toHaveBeenCalledTimes(9);
  });

  test('Should match snapshot', () => {
    // Arrange
    const fn = jest.fn();
    const { container } = render(
      <Router>
        <InputType
          type="any type"
          id="any id"
          nameField="any name"
          dataTestId="any testid"
          className="any className"
          placeholder="any placeholder"
          value="any value"
          min="any min"
          onChange={fn}
          onKeyDown={fn}
          autoComplete="on"
          aria-label="any id"
          aria-required
          required
          disabled
          defaultChecked
        />
      </Router>
    );

    // Act & Assert
    expect(container).toMatchSnapshot();
  });
});
