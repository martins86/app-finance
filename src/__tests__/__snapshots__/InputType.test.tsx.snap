// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <InputType /> Should match snapshot 1`] = `
<div>
  <input
    autocomplete="on"
    checked=""
    class="any className"
    data-testid="any testid"
    disabled=""
    id="any id"
    min="any min"
    name="any name"
    placeholder="any placeholder"
    required=""
    type="any type"
    value="any value"
  />
</div>
`;
